scripts/o3de/o3de/repo.py

```diff
@@ -339,9 +387,23 @@ def search_repo(manifest_json_data: dict,
     return search_o3de_object(manifest_json, o3de_object_uris, search_func)
 
 
+def search_o3de_repo_for_object(repo_json_data: dict, manifest_attribute:str, target_json_key:str, target_name: str):
+    o3de_objects = repo_json_data.get(manifest_attribute, [])
+    search_func = lambda repo_json_data: repo_json_data if repo_json_data.get(target_json_key, '') == target_name else None
+    for o3de_object in o3de_objects:
+        result_json_data = search_func(o3de_object)
+        if result_json_data:
+            return result_json_data
+    return None
```

Comment:
why do we need search_func? can we not do this?
```suggestion
    for o3de_object in o3de_objects:
        if o3de_object.get(target_json_key,'') == target_name:
            return o3de_object
    return None
```
or there may be a more Pythonic - but harder to read? version similar to:
```
    return data for data in o3de_objects if data.get(target_json_key,'') == target_name else None
```

Reply:
as for the one liner suggested, that is definitely harder to read. I'd go with the first suggestion you made